.github/workflows/robot/internal/leave/leave.go

```diff
@@ -0,0 +1,279 @@
+/*
+Copyright 2022 Gravitational, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package leave
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"log"
+	"net/http"
+	"net/url"
+	"path"
+	"strings"
+	"time"
+
+	"github.com/gravitational/trace"
+)
+
+type Config struct {
+	Ctx context.Context
+	// RipplingToken is the Rippling authentication token.
+	RipplingToken string
+	// Usernames is a JSON string of employees' Github
+	// usernames mapped to their full name.
+	Usernames string
+}
+
+type leave struct {
+	// onLeave is a map of employees who are on leave.
+	onLeave map[string]bool
+	// usernames is employees' GitHub usernames mapped to their
+	// full name.
+	usernames map[string]string
+}
+
+func New(c *Config) (Leave, error) {
+	if err := c.CheckAndSetDefaults(); err != nil {
+		return leave{}, nil
+	}
+	omit, err := getEmployeesOnLeave(c.Ctx, c.RipplingToken)
+	if err != nil {
+		return leave{}, trace.Wrap(err)
+	}
+	var employeeUsernames map[string]string
+	if err := json.Unmarshal([]byte(c.Usernames), &employeeUsernames); err != nil {
+		return leave{}, nil
+	}
+	return leave{onLeave: omit, usernames: employeeUsernames}, nil
+}
+
+func (c *Config) CheckAndSetDefaults() error {
+	if c.RipplingToken == "" {
+		return trace.BadParameter("missing parameter RipplingToken")
+	}
+	if c.Ctx == nil {
+		c.Ctx = context.Background()
+	}
+	if c.Usernames == "" {
+		return trace.BadParameter("missing parameter Usernames")
+	}
+	return nil
+}
+
+type Leave interface {
+	ShouldOmit(name string) bool
+}
+
+func (l leave) ShouldOmit(username string) bool {
+	fullname, ok := l.usernames[username]
+	if !ok {
+		return false
+	}
+	if _, ok := l.onLeave[fullname]; ok {
```

Comment:
Doesn't this have the same problem as before, i.e. employees may have the same `fullname` so this map may have a key conflict?